src/OptimalTransport.jl

```diff
@@ -715,33 +715,82 @@ Returns the entropically regularised barycenter of the `μ`, i.e. the histogram
 where ``\\operatorname{OT}_{ε}(\\mu, \\nu) = \\inf_{\\gamma \\Pi(\\mu, \\nu)} \\langle \\gamma, C \\rangle + \\varepsilon \\Omega(\\gamma)`` 
 is the entropic optimal transport loss with cost ``C`` and regularisation ``\\epsilon``.
 """
-function sinkhorn_barycenter(μ, C, ε, w; tol=1e-9, check_marginal_step=10, max_iter=1000)
-    sums = sum(μ; dims=1)
-    if !isapprox(extrema(sums)...)
-        throw(ArgumentError("Error: marginals are unbalanced"))
+function sinkhorn_barycenter(
+    μ,
+    C,
+    ε,
+    w;
+    tol=nothing,
+    atol=tol,
+    rtol=nothing,
+    check_marginal_step=nothing,
+    check_convergence=10,
+    max_iter=nothing,
+    maxiter=1_000,
+)
+    if tol !== nothing
+        Base.depwarn(
+            "keyword argument `tol` is deprecated, please use `atol` and `rtol`",
+            :sinkhorn_barycenter,
+        )
+    end
+    if check_marginal_step !== nothing
+        Base.depwarn(
+            "keyword argument `check_marginal_step` is deprecated, please use `check_convergence`",
+            :sinkhorn_barycenter,
+        )
     end
+    if max_iter !== nothing
+        Base.depwarn(
+            "keyword argument `max_iter` is deprecated, please use `maxiter`",
+            :sinkhorn_barycenter,
+        )
+    end
+    if !isapprox(extrema(sum(μ; dims=1))...)
+        throw(ArgumentError("Error: input marginals must have the same mass"))
+    end
+
+    T = float(Base.promote_eltype(μ, C))
```

Comment:
It is better to compute `K` first and then use
```suggestion
    T = float(Base.promote_eltype(μ, K))
```